### DISCUSSION TITLE
Any way to change IME ?

### DISCUSSION THREAD

**Discussion by zengfanfan** [microsoft/vscode-discussions#865] [Extension Development QnA] [answered]
I want to make a extension doing this:
1. when typing code, use pure english keyboard
2. when typing comment, use local IME

Is there any way to implement it?

---------------
I've thought about sending some key event, like Shift or Ctrl+Space, and found [this](https://github.com/microsoft/vscode/issues/104832). However, fake key-event is kind of dangerous behavior, and it's not supported yet.

**Comment by isidorn** [comment, marked as answer]
What exactly do you want to do in your extension? Change language? This is not supported, this is user initiated.

**Reply by zengfanfan** [reply]
yes.
will it be supported in the future?

**Reply by isidorn** [reply]
No.
